services/criterion/criterion.service.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService } = require('..')
+
+const IMPROVED = 'improved'
+
+function errorMessagesFor(notFoundMessage = 'status not found') {
+  return {
+    404: notFoundMessage,
+    422: notFoundMessage,
+  }
+}
+
+const schema = Joi.string().required()
+
+module.exports = class Criterion extends BaseJsonService {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get route() {
+    return {
+      base: 'criterion',
+      pattern: ':user/:repo',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Criterion',
+        namedParams: {
+          user: 'chmoder',
+          repo: 'data_vault',
+        },
+        staticPreview: this.render({ isImproved: true, status: IMPROVED }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'criterion' }
+  }
+
+  static render({ isImproved, status }) {
+    if (isImproved) {
```

Comment:
bit of a nit, but not clear to me why we're checking the status value twice. my sense is that there's some set of possible values (perhaps as narrow as `improved` and `regressed`), and that from a rendering perspective we want a `brightgreen` color if `status` was `improved`, otherwise `yellow`.

not a blocker for me, but i think this could be simplified as a `if (status === 'improved') { ....` and eliminating the param in the sig here and check and arg within `handle`

Reply:
Actually this is a great nit.  Since this is my first time in the project I used examples to make this badge.  I am glad you suggested the change.  At this time, there are only two "text" statuses, and 404 response coming from the API.  This is unneeded complexity and removing it, makes it more readable.